### ISSUE TITLE
MUI V5: Rating component does not support fractional choice in Playwright

### ISSUE THREAD

**Issue by tangentlin** [atomic-testing/atomic-testing#86]
When trying to set value fraction such as 3.5, Playwright e2e would not work properly.

**Comment by tangentlin** [comment]
### Confirmed still open — root cause verified, fix folded into #878

Empirically verified (jsdom + real-browser chromium, driving the actual `RatingDriver`) that this is **not** addressed by #876 and reproduces today:

- **jsdom**: `setValue(3.5)` → `getValue()===3.5` (passes — but this is a **false green**: jsdom has no layout/visibility model).
- **Real browser (chromium)**: `setValue(3.5)` **fails** — `RatingDriver.setValue` locates the `3.5` radio and clicks its `label[for=…]`, but that half-star label is `width:0; position:absolute; overflow:hidden`, so Playwright rejects it as *not visible* and the click times out. `document.elementFromPoint` at the label's center resolves to the **adjacent full-star label (value 4)**, so even a forced coordinate click would set the wrong value. Integer `setValue(4)` works (control), so the failure is fractional-specific.

**Root cause:** MUI `precision={0.5}` stacks a zero-width half-star label behind the full-star label; the driver's label-click `setValue` has no fractional handling and can't hit it in a real browser.

**Fix:** route `RatingDriver.setValue` through a coordinate-free `activate()` click on the visually-hidden `<input type=radio value="3.5">` directly (bypassing the label), or a sub-star geometry click. The `activate()` primitive is now scoped in **#878**, which also covers the Rating clear (#68) — so this issue is tracked there. (`RatingDriver` is identical across v5/v6/v7, so the fix applies to all three; V5/V6 e2e is separately blocked by #877.)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

**Comment by tangentlin** [comment]
Re-scoped from v5 to **v6/v7**. MUI 5 reached end of support on 2026-06-27 ([ADR-005](https://github.com/atomic-testing/atomic-testing/blob/main/agent-docs/adr/005-drop-mui-5-support.md)), but the Rating driver is shared across majors so this fractional-`setValue` gap still applies to v6/v7. The underlying fix — a coordinate-free `activate` primitive that clicks the hidden `<input type=radio value="3.5">` directly — is already designed in #878, which this issue depends on.
